pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -224,22 +224,41 @@ func GatherContainerImages(i *Gatherer) func() ([]record.Record, []error) {
 				if strings.HasPrefix(pod.Namespace, "openshift") && hasContainerInCrashloop(podPtr) {
 					records = append(records, record.Record{Name: fmt.Sprintf("config/pod/%s/%s", pod.Namespace, pod.Name), Item: PodAnonymizer{podPtr}})
 				} else if pod.Status.Phase == corev1.PodRunning {
-					for _, container := range pod.Spec.Containers {
-						imageURL := container.Image
-						urlHost, err := forceParseURLHost(imageURL)
-						if err != nil {
-							klog.Errorf("unable to parse container image URL: %v", err)
-							continue
-						}
-						if imageHostRegex.MatchString(urlHost) {
-							imgIndex, ok := image2idx[imageURL]
-							if !ok {
-								imgIndex = contInfo.Images.Add(imageURL)
-								image2idx[imageURL] = imgIndex
+					gatherImages := func(containers []corev1.ContainerStatus) {
```

Comment:
I know the readability suffers, but one of the variables (a property of the pod object) is being captured into the inline function. It could be done otherwise, but this results in the cleanest calls. Also, the output slice is captured in there. The calls don't look very nice when I put it out and considering that inline functions are very common in the I.O. code, I thought it wasn't that big of an issue.

Reply:
you can pass the variable as function parameter, I agree this is pretty long to be inline, but it is nit